packages/stream_chat/lib/src/api/channel.dart

```diff
@@ -1507,123 +1505,125 @@ class ChannelClientState {
+      .map((e) => _channel.client.state!.users[e.id] ?? e)
       .toList();
 
   /// Channel watchers list as a stream
-  Stream<List<User>> get watchersStream =>
-      CombineLatestStream.combine2<List<User>, Map<String, User>, List<User>>(
-        channelStateStream.map((cs) => cs.watchers),
-        _channel.client.state.usersStream,
-        (watchers, users) => watchers.map((e) => users[e.id] ?? e).toList(),
+  Stream<List<User>> get watchersStream => CombineLatestStream.combine2<
+          List<User>?, Map<String?, User?>, List<User>>(
+        channelStateStream.map((cs) => cs!.watchers),
+        _channel.client.state!.usersStream,
+        (watchers, users) => watchers!.map((e) => users[e.id] ?? e).toList(),
       );
 
   /// Channel read list
-  List<Read> get read => _channelState.read;
+  List<Read>? get read => _channelState!.read;
 
   /// Channel read list as a stream
-  Stream<List<Read>> get readStream => channelStateStream.map((cs) => cs.read);
+  Stream<List<Read>?> get readStream =>
+      channelStateStream.map((cs) => cs!.read);
 
   final BehaviorSubject<int> _unreadCountController = BehaviorSubject.seeded(0);
 
   /// Unread count getter as a stream
   Stream<int> get unreadCountStream => _unreadCountController.stream;
 
   /// Unread count getter
-  int get unreadCount => _unreadCountController.value;
+  int? get unreadCount => _unreadCountController.value;
 
   bool _countMessageAsUnread(Message message) {
     final userId = _channel.client.state?.user?.id;
-    final userIsMuted = _channel.client.state?.user?.mutes?.firstWhere(
-          (m) => m.user?.id == message.user.id,
-          orElse: () => null,
+    final userIsMuted = _channel.client.state?.user?.mutes.firstWhereOrNull(
+          (m) => m.user?.id == message.user!.id,
         ) !=
         null;
     return message.silent != true &&
         message.shadowed != true &&
-        message.user.id != userId &&
+        message.user!.id != userId &&
         !userIsMuted;
   }
 
   /// Update threads with updated information about messages
-  void updateThreadInfo(String parentId, List<Message> messages) {
-    final newThreads = Map<String, List<Message>>.from(threads);
+  void updateThreadInfo(String? parentId, List<Message>? messages) {
+    final newThreads = Map<String?, List<Message>?>.from(threads!);
 
     if (newThreads.containsKey(parentId)) {
       newThreads[parentId] = [
         ...newThreads[parentId]
-                ?.where(
-                    (newMessage) => !messages.any((m) => m.id == newMessage.id))
-                ?.toList() ??
+                ?.where((newMessage) =>
+                    !messages!.any((m) => m.id == newMessage.id))
+                .toList() ??
             [],
-        ...messages,
+        ...messages!,
       ];
 
-      newThreads[parentId].sort(_sortByCreatedAt);
+      newThreads[parentId]!
+          .sort(_sortByCreatedAt as int Function(Message, Message)?);
```

Comment:
I think the function param should be typed rather than casting it here.
```dart
int _sortByCreatedAt(Message a, Message b) =>
      a.createdAt.compareTo(b.createdAt);
```